tests/utils/utils.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals describe, it, expect */
+
+'use strict';
+
+var modules = bender.amd.require( 'utils' );
+
+describe( 'extend()', function() {
+	it( 'should extend and override', function() {
+		var utils = modules.utils;
+
+		var target = {
+			a: 1,
+			b: 2
+		};
+
+		var extensions = {
+			a: 'A',
+			c: 3,
+
+			// Extend by reference (no deep-copy).
+			obj: {
+				a: 1
+			},
+
+			// Extend by reference (no deep-copy).
+			arr: [ 1, 2 ],
+
+			// Extend by reference.
+			fn: function() {}
+		};
+
+		var ret = utils.extend( target, extensions );
+
+		expect( target.a ).to.equal( 'A' );
+		expect( target.b ).to.equal( 2 );
+		expect( target ).to.have.property( 'c' ).to.equal( 3 );
+
+		expect( target ).to.have.property( 'obj' ).to.equal( extensions.obj );
+		expect( target ).to.have.property( 'arr' ).to.equal( extensions.arr );
+		expect( target ).to.have.property( 'fn' ).to.equal( extensions.fn );
+
+		// "target" should be the return value.
+		expect( ret ).to.equal( target );
+	} );
+
+	it( 'should not be touched by non-objects', function() {
+		var utils = modules.utils;
+
+		var target = {
+			a: 1
+		};
+
+		expect( utils.extend( target, function() {} ) ).to.equal( target );
+		expect( utils.extend( target, 1 ) ).to.equal( target );
+		expect( utils.extend( target, 'a' ) ).to.equal( target );
+		expect( utils.extend( target, true ) ).to.equal( target );
+		expect( utils.extend( target, undefined ) ).to.equal( target );
+		expect( utils.extend( target, [] ) ).to.equal( target );
+		expect( utils.extend( target, Date.now() ) ).to.equal( target );
```

Comment:
`Date.now()` returns a number, so what's the point of this case since the number is being tested few lines earlier?